packages/vue-client/src/components/GameCreation/SFractionalConfigForm.vue

```diff
@@ -0,0 +1,64 @@
+<script setup lang="ts">
+import { BoardConfig, SFractionalConfig } from "@ogfcommunity/variants-shared";
+import BoardConfigForm from "./BoardConfigForms/BoardConfigForm.vue";
+
+const props = defineProps<{
+  initialConfig: SFractionalConfig;
+}>();
+
+const config: SFractionalConfig = {
+  ...props.initialConfig,
+  board: { ...props.initialConfig.board },
+  secondary_colors: [...props.initialConfig.secondary_colors],
+};
+
+const emit = defineEmits<{
+  (e: "configChanged", config: SFractionalConfig): void;
+}>();
+
+function emitConfigChange() {
+  emit("configChanged", config);
+}
+
+function setBoardConfig(boardConfig: BoardConfig): void {
+  config.board = boardConfig;
+  emitConfigChange();
+}
+
+/* TODO: Figure out how to make this work with vues reactivity
```

Comment:
I think you just need to wrap `config` in `reactive()`:

```
const config: SFractionalConfig = reactive({
  ...props.initialConfig,
  board: { ...props.initialConfig.board },
  secondary_colors: [...props.initialConfig.secondary_colors],
});
```

Vue is capable of detecting updates on reactive arrays!